cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/PostgresSqlStructuredTable.java

```diff
@@ -138,6 +130,93 @@ public void upsert(Collection<Field<?>> fields) throws InvalidFieldException, IO
     }
   }
 
+  @Override
+  public boolean compareAndSwap(Collection<Field<?>> keys, Field<?> oldValue, Field<?> newValue)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: CompareAndSwap with keys {}, oldValue {}, newValue {}", tableSchema.getTableId(), keys,
+              oldValue, newValue);
+    tableSchema.validatePrimaryKeys(keys.stream().map(Field::getName).collect(Collectors.toList()), false);
+    if (tableSchema.getType(oldValue.getName()) == null) {
```

Comment:
Would it be better to have validation done in a separate class? That way the two implementations can share certain validations.

Reply:
+1

Reply:
Filed JIRA to consolidate the field validations - CDAP-14833